File: src/Asm.MooBank.Data/Entities/AccountGroup/AccountGroup.cs
using System.ComponentModel.DataAnnotations.Schema;
using Asm.Domain;

namespace Asm.MooBank.Domain.Entities.AccountGroup;

[AggregateRoot]
public class AccountGroup : IIdentifiable<Guid>
{

    public AccountGroup()
    {
        Accounts = new HashSet<Account.Account>();
    }

    public Guid Id { get; set; }

    public string Name { get; set; }

    public string? Description { get; set; }

    public Guid OwnerId { get; set; }

    public bool ShowPosition { get; set; }

    public virtual AccountHolder.AccountHolder Owner { get; set; }

    [NotMapped]
    public virtual ICollection<Account.Account> Accounts { get; set; }

}

File: src/Asm.MooBank.Web.App/src/models/AccountGroup.ts
import { emptyGuid } from "@andrewmclachlan/mooapp";

export interface AccountGroup {
    id: string;
    name: string;
    description?: string;
    showPosition: boolean;
}

export const emptyAccountGroup : AccountGroup = {
    id: emptyGuid,
    name: "",
    showPosition: true,
}